packages/react/src/components/UIShell/SideNavItems.js

```diff
@@ -19,6 +21,13 @@ const SideNavItems = ({
 }) => {
   const className = cx([`${prefix}--side-nav__items`], customClassName);
   const childrenWithExpandedState = React.Children.map(children, child => {
+    if (
+      typeof child.type === 'string' ||
```

Comment:
Isn't the `typeof` the type always string? Maybe should just be `child.type`.

This sort of type inference is generally super fragile. What happens if someone uses a custom `SideNavItem` or otherwise wraps the item?

This whole boolean expression is really difficult to parse what the intended behavior is.

Reply:
no, type is string only if child is DOM element, not React Component, so if you put <a> tag directly as a child

Reply:
I agree, ultimately the issue lies in children that wrap DOM elements, like for example `UIShell/Link`, that component just passes through all props to React.createElement function.. thus my initial pr had a fix on Link element, I don't think that there is a way to determine at this level that your child is just passing all props to DOM element